packages/front-end/components/AuthorizedImage.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useEffect, useState, FC } from "react";
+import { FaExclamationTriangle } from "react-icons/fa";
+import { useAuth } from "@/services/auth";
+import {
+  getApiHost,
+  getGcsDomain,
+  getS3Domain,
+  usingFileProxy,
+} from "@/services/env";
+import LoadingSpinner from "./LoadingSpinner";
+
+interface AuthorizedImageProps extends React.HTMLProps<HTMLImageElement> {
+  imageCache?: Record<string, string>;
+}
+
+/**
+ * This component is used to display images that are stored in a private bucket.
+ * It will fetch the image from the backend and convert it to a blob url that can be displayed.
+ * It will also cache the image if imageCache is set so that it does not need to be fetched again.
+ * */
+const AuthorizedImage: FC<AuthorizedImageProps> = ({
+  imageCache = {},
+  src = "",
+  ...props
+}) => {
+  const [imageSrc, setImageSrc] = useState<string | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const { apiCall } = useAuth();
+
+  useEffect(() => {
+    const fetchData = async (src, apiUrl) => {
+      try {
+        const imageData: Blob = await apiCall(new URL(apiUrl).pathname);
+        const imageUrl = URL.createObjectURL(imageData);
+        imageCache[src] = imageUrl;
+        setImageSrc(imageUrl);
+      } catch (error) {
+        setError(error.message);
+      }
+    };
+
+    if (imageCache[src]) {
+      // Images in the cache do not need to be fetched again
+      setImageSrc(imageCache[src]);
+    } else if (
+      usingFileProxy() &&
+      getGcsDomain() &&
+      src.startsWith(getGcsDomain())
```

Comment:
For this (and for the similar s3 check below) I'm having trouble understanding when the `src` property will be set to an S3 url and not a GB host. When the user uploads a screenshot, would we not update the variation with the GB proxy URL? Is this meant for existing customers to accomplish a backfill?